cub/agent/agent_radix_sort_downsweep.cuh

```diff
@@ -193,34 +193,42 @@ struct AgentRadixSortDownsweep
     _TempStorage    &temp_storage;
 
     // Input and output device pointers
-    KeysItr         d_keys_in;
-    ValuesItr       d_values_in;
-    UnsignedBits    *d_keys_out;
-    ValueT          *d_values_out;
+    KeysItr            d_keys_in;
+    ValuesItr          d_values_in;
+    bit_ordered_type  *d_keys_out;
+    ValueT            *d_values_out;
 
     // The global scatter base offset for each digit (valid in the first RADIX_DIGITS threads)
-    OffsetT         bin_offset[BINS_TRACKED_PER_THREAD];
+    OffsetT bin_offset[BINS_TRACKED_PER_THREAD];
 
-    // Digit extractor
-    DigitExtractorT digit_extractor;
+    std::uint32_t current_bit; 
+    std::uint32_t num_bits;
 
     // Whether to short-cirucit
-    int             short_circuit;
+    int short_circuit;
```

Comment:
Why is this an `int `rather than a `bool`?

Reply:
I don't know why author preferred int here. I also don't think we should spend time verifying that making it bool doesn't affect perf as part of this PR.